### ISSUE TITLE
org-mode compatibility

### ISSUE THREAD

**Issue by mooseyboots** [purcell/page-break-lines#35]
i know it's probably really meant for prog-modes, but i just got started on a text project converted from pdf, and all page breaks are rendered as `^L`.

when i insert an org heading, any horizontal lines following it become, i think, infinitely long, or very long, and if you try to move point past one, point goes way back up the buffer.

should this mode work in org (and so it is maybe something with my set up) or is it simply a no-go?

the problem is still there for me if i set a max width for page-break-lines, and also with any variable pitch modes disabled. (not sure what else it could be interacting with.)

**Comment by purcell** [comment]
> when i insert an org heading, any horizontal lines following it become, i think, infinitely long, or very long, and if you try to move point past one, point goes way back up the buffer.
> 
> should this mode work in org (and so it is maybe something with my set up) or is it simply a no-go?

It should work. I'm not completely clear on the steps to reproduce. Could you perhaps paste a short example snippet of org content that is problematic for you?

**Comment by mooseyboots** [comment]
ah, i just realized, sorry, it only mucks up when `visual-line-mode` is enabled. maybe there's no way around that?

**Comment by basaran** [comment]
I use auto-fill-mode for my org files and set everything to a narrow width of 60 chars.  You can perhaps try shadowing the `page-break-lines--update-display-table` function.

For proof of concept, replace `make-list width glyph` with 
`make-list fill-column glyph`.

and set the fill column to whatever you like. Thank you `purcell` :)

**Comment by SterlingHooten** [comment]
I'm also having this issue, and it seems to be a combination of having both `visual-line-mode` and `org-mode` enabled together.

**Comment by purcell** [comment]
If I enable `visual-line-mode` in an `org-mode` buffer and play around with adding page breaks, I really don't see any misbehaviour. At least in Emacs 28.2. Could you give me your Emacs version and perhaps a few steps to reproduce, @SterlingHooten?

**Comment by tshu-w** [comment]
For me, it is conflicting if I turn on `truncate-lines`,
```emacs-lisp
;;; emacs -Q -l test.el inbox.org

(setq package-archives '(("melpa"        . "https://melpa.org/packages/")
                         ("gnu"          . "https://elpa.gnu.org/packages/")
                         ("nongnu"       . "https://elpa.nongnu.org/nongnu/")))

(package-initialize)

(use-package org
  :ensure t
  :init
  (setq org-startup-indented t)
  (add-hook 'org-mode-hook
            (defun init-org-mode ()
              "Stuff to do when opening `org-mode' files."
              (setq truncate-lines nil))))

(use-package page-break-lines
  :ensure t
  :init
  (global-page-break-lines-mode))
```

```org-mode
#+title: Inbox

* 
This is a test.  ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff

```

**Comment by purcell** [comment]
Right, yes, I can reproduce this by enabling `org-indent-mode` and disabling `truncate-lines`. It even hangs Emacs somewhat, so perhaps the line ends up incredibly long, unsure. I'll try to find some time to debug this, but I have no idea what magic `org-indent-mode` uses.

**Comment by purcell** [comment]
I had a look at this, changing `page-break-lines` to add some debug info as follows:
```diff
diff --git a/page-break-lines.el b/page-break-lines.el
index 2809579..3cb5c93 100644
--- a/page-break-lines.el
+++ b/page-break-lines.el
@@ -135,6 +135,7 @@ its display table will be modified as necessary."
                               width))
                      (glyph (make-glyph-code page-break-lines-char 'page-break-lines))
                      (new-display-entry (vconcat (make-list width glyph))))
+                (message "PBL: width=%d, max-chars-per-line=%d, rel-width=%d, entry-length=%d" width (window-max-chars-per-line) char-relative-width (length new-display-entry))
                 (unless (equal new-display-entry (elt buffer-display-table ?\^L))
                   (aset buffer-display-table ?\^L new-display-entry)))))
         (when (and (apply 'derived-mode-p page-break-lines-modes)
```
In the test scenario, this prints that the new display entry for `^L` is of roughly the expected length, probably a few characters too long (due to org's indent), and `M-x describe-char` at the beginning of the line confirms the display table entry length. But nonetheless a much longer wrapped line is displayed, as in your screenshot. This is a mystery to me, I'm afraid. I suspect there's some miscalculation happening in Emacs' own display code when `truncate-lines` is off, or in `org-indent`, though I don't see anything obvious there.

**Comment by tshu-w** [comment]
> In the test scenario, this prints that the new display entry for `^L` is of roughly the expected length, probably a few characters too long (due to org's indent), and `M-x describe-char` at the beginning of the line confirms the display table entry length.

I tried the similar thing and got the same result as you. Maybe we need to report this to the upstream, but I'm not sure.
